sdk/eventhub/Azure.Messaging.EventHubs.Processor/src/EventProcessorClient.cs

```diff
@@ -742,7 +741,10 @@ public virtual async Task StopProcessingAsync(CancellationToken cancellationToke
 
             try
             {
-                await StorageManager.Value.UpdateCheckpointAsync(checkpoint).ConfigureAwait(false);
+                // TODO: apply retry policy here.  Do not call error handler in case of failure and notify
```

Comment:
If you're looking for an example of the pattern that we've applied elsewhere in the library for retries, the [Producer](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/eventhub/Azure.Messaging.EventHubs/src/Amqp/AmqpProducer.cs#L267) implementation is a reasonably uncluttered example.

Reply:
Should the retry logic be in the storage manager?   That would seem to be a reasonable idea for it to be consistently applied and would match how the processor interacts with other clients that it delegates to (like the consumer).